tests/ui/README.md

```diff
@@ -0,0 +1,104 @@
+them in the future, it would be a good idea to create a fork of this repository
+first, and then clone that. GitHub also has great instructions for
+[forking a repository][git-fork].
+
+### Server Install
+
+Follow the instructions found [here][addons-server-docs].
+
+### Run the tests
+
+Included in the docker-compose file is an image containing firefox nightly. [tox][Tox]
+is our test environment manager and [pytest][pytest] is the test runner.
+
+To run the tests, execute the command below:
+```sh
+docker-compose exec --user root selenium-firefox tox -e ui-tests
+```
+
+### Adding a test
+
+The tests are written in python using a POM, or Page Object Model. The plugin we use for this is called [pypom][pypom]. Please read the documentation there for good examples
+on how to use the Page Object Model when writing tests.
+
+The pytest plugin that we use for running tests has a number of advanced command
+line options available too. The full documentation for the plugin can be found [here][pytest-selenium].
+
+## Additional Information
+
+The tests run against the newest version of the [AMO][amo] frontend using a docker image provided by [addons-frontend][addons-frontend]. You can view the frontend after the build has been completed at ```localhost:4000```.
+
+### Watching a test run
+
+The tests are run on a live version of firefox, but they are run headless. To access the container where the tests are run to view them follow these steps:
+
+IMPORTANT: Please comment out this line within the ```tests/ui/conftest.py``` file if you would like to view or debug the tests.
+```sh
+firefox_options.add_argument('-headless')
+```
+
+1. Make sure the container is running:
+```sh
+docker-compose ps
+```
+
+2. Copy the port that is forwarded for the ```selenium-firefox``` image:
+```sh
+0.0.0.0:32771->5900/tcp
+```
+Note: Your port may not match what is seen here.
+
+You will want to copy what ever IP address and port is before the ```->5900/tcp```.
+
+3. Open your favorite VNC viewer and type in, or paste that address.
+4. The password is ```secret```.
+5. The viewer should open a window with a Ubuntu logo. If that happens you are connected to the ```selenium-firefox``` image and if you start the test, you should see a firefox window open and the tests running.
+
+### Firefox setup
+
+The preferences used to setup firefox are here:
+```sh
+    firefox_options.set_preference(
+        'extensions.install.requireBuiltInCerts', False)
+    firefox_options.set_preference('xpinstall.signatures.required', False)
+    firefox_options.set_preference('extensions.webapi.testing', True)
+    firefox_options.set_preference('ui.popup.disable_autohide', True)
+    firefox_options.add_argument('-foreground')
+    firefox_options.add_argument('-headless')
+    firefox_options.log.level = 'trace'
+```
+These shouldn't need to be touched as they allow for unsigned addon installation as well as
+disabling the autohide function and setting the firefox browser to run headless.
```

Comment:
Maybe link to GitHub or a file-path where these options can be found / modified.